std/datetime/mod.ts

```diff
@@ -150,3 +149,133 @@ export function toIMF(date: Date): string {
     months[date.getUTCMonth()]
   } ${y} ${h}:${min}:${s} GMT`;
 }
+
+/**
+ * Check given year is a leap year or not.
+ * @param year year in number or Date format
+ */
+export function isLeap(year: Date | number): boolean {
+  const yearNumber = year instanceof Date ? year.getFullYear() : year;
+  return (
+    !(yearNumber % 400) || (yearNumber % 4 === 0 && yearNumber % 100 !== 0)
+  );
+}
+
+export type Unit =
+  | "miliseconds"
+  | "seconds"
+  | "minutes"
+  | "hours"
+  | "days"
+  | "weeks"
+  | "months"
+  | "quarters"
+  | "years";
+
+export interface DifferenceFormat {
+  miliseconds?: number;
+  seconds?: number;
+  minutes?: number;
+  hours?: number;
+  days?: number;
+  weeks?: number;
+  months?: number;
+  quarters?: number;
+  years?: number;
+}
```

Comment:
these two types should still be inextricably linked so that if one changes, the other does too. You can use a Partial Record here: 

```suggestion
export type Unit =
  | "miliseconds"
  | "seconds"
  | "minutes"
  | "hours"
  | "days"
  | "weeks"
  | "months"
  | "quarters"
  | "years";

export type DifferenceFormat = Partial<Record<Unit, number>>
```